File: frontend/src/microservices/train/objects/TrainSuggestionsPlannerResponse.ts
// 本文件定义 train 模块的 `TrainSuggestionsPlannerResponse`，用于表示单条火车搜索建议并提供 JSON 编解码。

export type TrainSuggestionsPlannerResponse = {
  resourceType: string
  value: string
  title: string
  subtitle: string
}
export const trainSuggestionsPlannerResponseFromJson = (json: string): TrainSuggestionsPlannerResponse =>
  JSON.parse(json) as TrainSuggestionsPlannerResponse

export const trainSuggestionsPlannerResponseToJson = (value: TrainSuggestionsPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/train/objects/TrainSuggestionsPlannerResponse.scala
package com.typesafe.travel.train.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class TrainSuggestionsPlannerResponse(resourceType: String, value: String, title: String, subtitle: String)
object TrainSuggestionsPlannerResponse:
  given sourceEncoder: Encoder[TrainSuggestionsPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[TrainSuggestionsPlannerResponse] = deriveDecoder
